docs/technology.md

```diff
@@ -43,3 +43,119 @@ The slowest step in the pipeline is the graph database writer.
 While this is happening, python can suspend the task waiting on the database allowing the other steps to continue processing.
 This allows for the extractor and interpreter to continue processing data and not be constrained by the database writer.
+
+## Migrations 
+
+Nodestream from versions `0.1`` through `0.10`` automatically requested the creation of indexes and constraints on the database. 
+This was done by introspecting the schema of the entire project and generating the appropriate queries to create the indexes and constraints.
+This was a very powerful feature but it had a few drawbacks:
+- **It was redundant.** The same indexes and constraints were being created with `IF NOT EXISTS` clauses every time the pipeline was run.
+- **It was slow.** The queries were being executed serially and the pipeline was locked until they were all completed.
+- **It was error prone.** If the database was not in a state that allowed for the creation of the indexes and constraints, the pipeline would fail.
+- **It was high friction.** There was no way to refactor the database without manual intervention. If the schema changed, the pipeline would fail and the user would have to manually remove the indexes, constraints, and sometimes data before running the pipeline again.
+
+To address these issues, nodestream 0.11 introduced the concept of migrations.
+Migrations are a way of encapsulating changes to the database schema in a way that can be applied incrementally. 
+Conceptually, they are similar to the migrations in the [Django](https://docs.djangoproject.com/en/5.0/topics/migrations/), [Rails](https://guides.rubyonrails.org/v3.2/migrations.html), [Neo4j Migrations](https://neo4j.com/labs/neo4j-migrations/2.0/), and [Flyway](https://documentation.red-gate.com/fd/migrations-184127470.html) frameworks.
+
+
+### Scope
+
+Some changes to the database are tricier to deal with than others, for the 0.11 release, the following changes are supported:
+- **Creating indexes.** Creating indexes on node/relationship properties.
+- **Droping indexes.** Dropping indexes on node/relationship properties.
+- **Renaming indexes.** Renaming indexes on node/relationship properties (Drop and create).
+- **Creating Node / Property Types.** Registering new node and property types with the database and creating a constraint on the node type.
+- **Renaming Node / Relationship Types** Renaming a node or relationship type.
+- **Deleting Node / Relationship Types** Deleting a node or relationship type.
+- **Adding Properties** Adding a property on a node or relationship type with a default value.
+- **Renaming Properties** Renaming a property on a node or relationship type.
+- **Deleting Properties** Deleting a property on a node or relationship type.
+- **Extending Keys** Adding a property to a key constraint.
+- **Renaming Keys** Renaming a key constraint property. 
+
+Notably, some use cases that are not supported:
```

Comment:
Add a note about rolling back migrations. This is out of scope because we 
1.) aren't losing features because we don't have this today. 
2.) this adds additional complexity and requires further study.